src/utils/listener/block_downloader.go

```diff
@@ -196,6 +196,7 @@ func (self *BlockDownloader) run() error {
 		}
 	}
 
+	self.Stop()
```

Comment:
I don't think this is the right place to stop. If the intention is to stop the application after the last block is processed maybe it's better to create a task that calls Controller.Stop() after the last block is processed?

Reply:
done